### ISSUE TITLE
Building BOINC exits with an error on aarch64 with autoconf 2.71

### ISSUE THREAD

**Issue by idunno101** [BOINC/boinc#4568]
**Describe the bug**
When running the configure script on an aarch64 machine with autoconf 2.71 the script exits with an error and the build fails.
``` 
checking for x86intrin.h... no
checking for pmmintrin.h... no
checking for xmmintrin.h... no
checking for emmintrin.h... no
checking for immintrin.h... no
checking for avxintrin.h... no
checking for gcc options needed to detect all undeclared functions... cannot detect
configure: error: in `/run/build/boinc':
configure: error: cannot make gcc report undeclared builtins
See `config.log' for more details
```
config.log shows:
```
c++: error: unrecognized command line option '-mavx'
```

With autoconf 2.69 there's no error like that, and the build completes.
```
checking x86intrin.h usability... no
checking x86intrin.h presence... no
checking for x86intrin.h... no
checking pmmintrin.h usability... no
checking pmmintrin.h presence... no
checking for pmmintrin.h... no
checking xmmintrin.h usability... no
checking xmmintrin.h presence... no
checking for xmmintrin.h... no
checking emmintrin.h usability... no
checking emmintrin.h presence... no
checking for emmintrin.h... no
checking immintrin.h usability... no
checking immintrin.h presence... no
checking for immintrin.h... no
checking avxintrin.h usability... no
checking avxintrin.h presence... no
checking for avxintrin.h... no
checking whether _xgetbv is declared... no
checking whether xgetbv is declared... no
```

**Steps To Reproduce**
1. Use autoconf 2.71
2. Build the BOINC client on a linux aarch64 machine

**Expected behavior**
Expected to complete the build.

**System Information**
 - OS: Debian Sid
 - BOINC Version: 7.18.1 

**Additional context**
If I remove '-mavx' from 
https://github.com/BOINC/boinc/blob/b49adfb118211e11c719766c0d71e7bdfe7f3363/configure.ac#L695-L698
then the configure script successfully completes.

**Comment by danc86** [comment]
FYI Fedora is carrying a patch for this, which looks like a reasonable solution:  
https://src.fedoraproject.org/rpms/boinc-client/blob/rawhide/f/boinc-client-7.18-AC_CHECK_DECLS-change.patch

From: https://www.spinics.net/lists/fedora-devel/msg300012.html

**Comment by AenBleidd** [comment]
@danc86, thanks for your comment. You can go ahead and create a PR with this patch.

**Comment by AenBleidd** [comment]
Closed as a duplicate of #4736
